src/iarc7_motion/motion_planner.py

```diff
@@ -77,6 +90,28 @@ def run(self):
 
             rate.sleep()
 
+    def handle_ground_interaction_done(self, status, result):
+        if self._ground_interaction_task_callback is not None:
+            try:
+                self._ground_interaction_task_callback(status, result)
+            except Exception as e:
+                rospy.logerr('Exception sending result using ground interaction done cb')
+                rospy.logerr(str(e))
+                rospy.logerr(traceback.format_exc())
+                rospy.logwarn('Motion planner aborted task')
+                self._action_server.set_aborted()
+                self._task = None
+            self._ground_interaction_task_callback = None
+        else:
+            rospy.logerr('Ground interaction done callback received with no task callback available')
+
+    def _handle_ground_interaction_command(self, ground_interaction_command):
+        self._ground_interaction_task_callback = ground_interaction_command.completion_callback
```

Comment:
We should check here that we aren't overwriting a pending callback (what if the task requests this again before the first one finishes?)